File: two_crystal_ball/two_crystal_ball.js
function two_crystal_ball(breaks) {
    let jumpAmount = Math.floor(Math.sqrt(breaks.length));

    let i = jumpAmount;
    for(; i < breaks.length; i += jumpAmount) {
        if(breaks[i]) {
            break;
        }
    }

    i -= jumpAmount;
    for(; i < breaks.length; i++){
        if(breaks[i]) {
            return i;
        }
    }

    return -1;
}

console.log(two_crystal_ball([false, false, false, true, true]))
console.log(two_crystal_ball([false, false, false]))
console.log(two_crystal_ball([false, false, false, false, false, false, true, true]))

File: two_crystal_ball/two_crystal_ball.py
from typing import List
import math

def two_crystal_ball(breaks: List[bool]) -> int:
    jumpAmount = math.floor(math.sqrt(len(breaks)))

    i = jumpAmount
    while(i < len(breaks)):
        if(breaks[i]):
            break
        i += jumpAmount
    
    i -= jumpAmount
    while(i < len(breaks)):
        if(breaks[i]):
            return i
        i += 1
    
    return -1

print(two_crystal_ball([False, False, False, True, True]))
print(two_crystal_ball([False, False, False]))
print(two_crystal_ball([False, False, False, False, False, False, True, True]))